test/e2e_tests/specs/CriticalFlow/messagesInGroups-TC-8751.spec.ts

```diff
@@ -0,0 +1,194 @@
+      const filePath = await userBModals.detailViewModal().downloadAsset();
+      const downloadQRCodeValue = await getLocalQRCodeValue(filePath);
+      const localQRCodeValue = await getLocalQRCodeValue(imageFilePath);
+      expect(downloadQRCodeValue).toBe(localQRCodeValue);
+    });
+
+    await test.step('User B reacts to A’s image', async () => {
+      await userBModals.detailViewModal().givePlusOneReaction();
+      await userBModals.detailViewModal().closeModal();
+      expect(await userBPages.conversation().isPlusOneReactionVisible()).toBeTruthy();
+    });
+
+    await test.step('User A can see the reaction', async () => {
+      // TODO: Bug [WPB-18226], remove this when fixed
+      await userAPageManager.refreshPage({waitUntil: 'load'});
+
+      expect(await userAPages.conversation().isPlusOneReactionVisible()).toBeTruthy();
+    });
+
+    await test.step('User A sends video message', async () => {
+      await userAComponents.inputBarControls().clickShareFile(videoFilePath);
+      expect(await userAPages.conversation().isVideoMessageVisible()).toBeTruthy();
+    });
+
+    await test.step('User B can play the message', async () => {
+      await userBPages.conversation().playVideo();
+      // Wait for 5 seconds to ensure video starts playing
+      await userBPages.conversation().page.waitForTimeout(5000);
+      // ToDO: Bug -> Video is not loaded from the server, so we cannot check if it is playing
+    });
+
+    await test.step('User A sends audio file', async () => {
+      await userAComponents.inputBarControls().clickShareFile(audioFilePath);
+      expect(await userAPages.conversation().isAudioMessageVisible()).toBeTruthy();
+    });
+
+    await test.step('User B can play the file', async () => {
+      await userBPages.conversation().playAudio();
+      // Wait for 5 seconds to ensure audio starts playing
+      await userBPages.conversation().page.waitForTimeout(5000);
+      expect(await userBPages.conversation().isAudioPlaying()).toBeTruthy();
+    });
+
+    await test.step('User A sends a quick (10 sec) self deleting message', async () => {
+      await userAComponents.inputBarControls().setEphemeralTimerTo('10 seconds');
+      await userAPages.conversation().sendMessage(selfDestructMessageText);
+      expect(await userAPages.conversation().isMessageVisible(selfDestructMessageText)).toBeTruthy();
+    });
+
+    await test.step('User B sees the message', async () => {
+      expect(await userBPages.conversation().isMessageVisible(selfDestructMessageText)).toBeTruthy();
+    });
+
+    await test.step('User B waits 10 seconds', async () => {
+      await userBPages.conversation().page.waitForTimeout(11_000);
+    });
+
+    await test.step('Both users see the message as removed', async () => {
+      expect(await userBPages.conversation().isMessageVisible(selfDestructMessageText)).toBeFalsy();
+      expect(await userAPages.conversation().isMessageVisible(selfDestructMessageText)).toBeFalsy();
```

Comment:
For falsy condition, you don't have to wait for visibility.
```suggestion
      expect(await userBPages.conversation().isMessageVisible(selfDestructMessageText, false)).toBeFalsy();
      expect(await userAPages.conversation().isMessageVisible(selfDestructMessageText, false)).toBeFalsy();
```

Reply:
Thanks, I'll apply these changes